### PR TITLE
[new] [author] add sanctuary

### PR DESCRIPTION
**Pull request by davidchambers**
:raising_hand_woman: _I would appreciate help with this pull request as it is my first contribution to this project. :)_

## Profile of the lib

* Git repository (required): <https://github.com/sanctuary-js/sanctuary>
* Official website (optional, not the repository): <https://sanctuary.js.org/>
* NPM package url (optional): <https://www.npmjs.com/package/sanctuary>
* License and its reference: <https://github.com/sanctuary-js/sanctuary/blob/master/LICENSE> (MIT)
* GitHub / Bitbucket popularity (required):
  * Count of watchers: 58
  * Count of stars: 2300
  * Count of forks: 94
* NPM download stats (optional):
  * Downloads in the last day:
  * Downloads in the last week:
  * Downloads in the last month:

## Essential checklist

* [x] I'm the author of this library
  * [ ] I would like to add link to the page of this library on CDNJS on website / readme
* [x] This lib was not found on cdnjs repo
* [x] Doesn't already exist / has duplicate issue and PR
* [x] The lib has notable popularity
  * [x] More than 200 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
  * [x] More than 800 downloads stats per month on npm registry
* [x] Project has public repository on famous online hosting platform (or been hosted on npm)

## Auto-update checklist

* [x] Has valid tags for each versions (for git auto-update)
* [x] Auto-update setup
* [ ] Auto-update target/source is valid.
* [ ] Auto-update filemap is correct.

## Git commit checklist

* [x] The first line of commit message is less then 50 chars; clean, clear and easy to understand.
* [x] The parent of the commit(s) in the PR is not older than 3 days.
* [x] Pull request is sent from a non-master branch with a meaningful name.
* [x] Separate unrelated changes into different commits.
* [x] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
* [x] Close corresponding issue in commit message
* [x] Mention related issue(s), people in commit message, comment.

### CODE REVIEW

File: ajax/libs/sanctuary/package.json [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  },
+  "dependencies": {
+    "sanctuary-def": "0.20.1",
+    "sanctuary-either": "1.2.0",
+    "sanctuary-maybe": "1.2.0",
+    "sanctuary-pair": "1.2.0",
+    "sanctuary-show": "1.0.0",
+    "sanctuary-type-classes": "11.0.0",
+    "sanctuary-type-identifiers": "2.0.1"
+  },
```
Comment (davidchambers) [inline ajax/libs/sanctuary/package.json]: Dependencies are subject to change from one version to the next. Can `dependencies` safely be omitted?

File: ajax/libs/sanctuary/package.json [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  "devDependencies": {
+    "fantasy-land": "4.0.1",
+    "jsverify": "0.8.x",
+    "sanctuary-descending": "1.2.0",
+    "sanctuary-identity": "1.2.x",
+    "sanctuary-scripts": "4.0.x"
+  },
```
Comment (davidchambers) [inline ajax/libs/sanctuary/package.json]: Can `devDependencies` safely be omitted?

File: ajax/libs/sanctuary/package.json [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  "files": [
+    "/LICENSE",
+    "/README.md",
+    "/index.js",
+    "/package.json"
+  ],
+  "filename": "bundle.min.js",
```
Comment (davidchambers) [inline ajax/libs/sanctuary/package.json]: Is this the correct value for `filename`? The (unminified) file in the repository is located at __dist/bundle.js__. My assumption is that `filename` must correspond to the name of the source file. Is this correct?

**Comment by davidchambers** [comment]
I have three questions.

> I would like to add link to the page of this library on CDNJS on website / readme

I'm having trouble parsing this sentence. What does it mean?

> Auto-update target/source is valid.

Is there some way for me to verify this?

> Auto-update filemap is correct.

Is there some way for me to verify this?

File: ajax/libs/sanctuary/package.json (hunk 1 above)
Comment (MattIPv4) [reply ajax/libs/sanctuary/package.json]: Yup.

File: ajax/libs/sanctuary/package.json (hunk 2 above)
Comment (MattIPv4) [reply ajax/libs/sanctuary/package.json]: Yup.

File: ajax/libs/sanctuary/package.json [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    "sanctuary-type-identifiers": "2.0.1"
+  },
+  "devDependencies": {
+    "fantasy-land": "4.0.1",
+    "jsverify": "0.8.x",
+    "sanctuary-descending": "1.2.0",
+    "sanctuary-identity": "1.2.x",
+    "sanctuary-scripts": "4.0.x"
+  },
+  "files": [
```
Comment (MattIPv4) [inline ajax/libs/sanctuary/package.json]: Files array isn't needed either.

File: ajax/libs/sanctuary/package.json (hunk 3 above)
Comment (MattIPv4) [reply ajax/libs/sanctuary/package.json]: As you've only specified `bundle.js` in the file-map, `filename` should be that. `filename` is the file that cdnjs will recommend as the default to use for the library.

Comment (davidchambers) [reply ajax/libs/sanctuary/package.json]: Thank you for your feedback, @MattIPv4!

> Note that you should have a `filename` to point to the main file of a lib, if that file is not minified, please still use `filename.min.js` or `filename.min.css` structure naming, we'll do the minify job.

Your suggestion appears to be at odds with the recommendation above, from __CONTRIBUTING.md__. Please confirm that `"bundle.js"` is the correct value.

File: ajax/libs/sanctuary/package.json (hunk 1 above)
Comment (davidchambers) [reply ajax/libs/sanctuary/package.json]: Thanks. I have removed `dependencies`. :)

File: ajax/libs/sanctuary/package.json (hunk 2 above)
Comment (davidchambers) [reply ajax/libs/sanctuary/package.json]: Thanks. I have removed `devDependencies`. :)

File: ajax/libs/sanctuary/package.json (hunk 4 above)
Comment (davidchambers) [reply ajax/libs/sanctuary/package.json]: Thanks. I have removed `files`. :)

File: ajax/libs/sanctuary/package.json (hunk 3 above)
Comment (MattIPv4) [reply ajax/libs/sanctuary/package.json]: I'm not aware of any auto-minification we have running anymore. I'd suggest including your own distributed min file in the file-map if you have it. You are then welcome to choose either as the default file :)

Comment (davidchambers) [reply ajax/libs/sanctuary/package.json]: I have changed `filename` to `"bundle.js"`, and have opened #13665 to update __CONTRIBUTING.md__. I believe I have addressed all your feedback, @MattIPv4. :)

**Comment by davidchambers** [comment]
Please let me know if there is anything more I can do here. :)

**Comment by davidchambers** [comment]
Could this pull request be merged, @MattIPv4?

**Comment by MattIPv4** [comment]
Merged, though I'm not sure the bot is currently detecting new package.json files, so it might not get populated with files for now.

**Comment by davidchambers** [comment]
> Merged, though I'm not sure the bot is currently detecting new package.json files, so it might not get populated with files for now.

@MattIPv4, is there anything we can do to get Sanctuary to appear on the website?

**Pull request merged**